src/libs/ajax.js

```diff
@@ -655,6 +658,34 @@ const Buckets = signal => ({
 })
 
 
+const GoogleBilling = signal => ({
+  listAccounts: async () => {
+    if (!hasBillingScope()) {
+      reportError('Terra does not have access to query your Google Cloud Billing information')
+    }
+    const response = await fetchGoogleBilling('billingAccounts', _.merge(authOpts(), { signal }))
+    const json = await response.json()
+    return json.billingAccounts
+  },
+
+  listProjectNames: async billingAccountName => {
+    try {
+      const response = await fetchGoogleBilling(`${billingAccountName}/projects`, _.merge(authOpts(), { signal }))
+      const json = await response.json()
+      return _.map('projectId', json.projectBillingInfo)
+    } catch (errorResponse) {
+      // This might happen if the user has permission to create projects with this account
+      // but is missing the billing.resourceAssociations.list permission.
+      if (errorResponse.status === 403) {
+        return []
```

Comment:
Should we return something more descriptive in this error state?

Reply:
This isn't really an error state. This translates the `403 Forbidden` response (due to lack of permissions) to "no data". The comment above is intended to communicate that. I can make that more explicit. I was intentionally a little bit aloof about it because I don't know what leads to this state, or if its even really a valid state to be in or just something I came across with dev data. What I do know is that I want to do this translation.

However, maybe the "I" in that statement is the caller of this function rather than the provider of this API. Moving this out to the caller would give the caller more flexibility about how to interpret this result, which might turn out to be important. I'll make that change unless anyone objects.